test_requirements.txt

```diff
@@ -1,4 +1,5 @@
 pytest>=2.8.7,<4.0.0
 pytest-cov<=2.6.0
 mock==2.0.0
-coverage
\ No newline at end of file
+coverage
```

Comment:
Maybe we should move these requirements into the `setup.py` `setup()`'s `extras_require` kwarg. They seem like tack-on extra functionality / features so they meet the use case of that kwarg.

It could be implemented something like this:
```python
setup(
    # other code....
    extras_require={
        "patch_filtering": [
            "whatthepatch",
        ],
        "coverage_filtering": [
            "coverage",
        ],
    },
)
```
That way if someone wants the enable the coverage filtering they could install it via pip with:

```shell
pip install mutmut["coverage_filtering"]
```

And if they wanted to enable patch filtering they could install via pip with:
```shell
pip install mutmut["patch_filtering"]
```

This should help limit core dependencies and lower the chance of dependency conflict with other peoples projects.

Reply:
Cool. Didn't know about that. I'll punt on this for now and we'll do this in a new PR after this lands.

Reply:
Yeah, that would be cool.